### PR TITLE
Add BMP280/BME280 I2C barometer support

### PR DESCRIPTION
**Pull request by CapnBry**
Adds a new type of barometer supported for Vario / Barometric Altitude telemetry, the Bosch BMP280 or BME280 (same except with unused humidity measurement). ESP8266 unified targets also can use barometers now as well, although only PWM RXes allow the pin function to be assigned to I2C currently.

It also adds BMP085 support but the code is commented out, because I am dumb and thought that's what I had. BMP085 is really old so I don't think we need to support it, but I wrote all the dang code so I left it but commented out.

### BMP280
I'd never heard of an SPL06 barometer until the first vario device arrived and I had to write our Vario implementation with it. The BMP280 has been much more commonly used (although the DPS310 may be #1 now) and is available in breakouts for users to add themselves.

I've used the same parameters used by iNav for its BMP280 device, resulting in a 23.025ms measurement period, but the values are only reported every 4th interval giving a ~91ms telemetry reporting period (~10.9Hz). The extra measurements are just used to feed the BMP's internal IIR filter.
* Pressure: 8x oversampling
* Temperature: 1x oversampling
* Filter coefficient: 8

### Changes to devBaro
Because the BMP280 measures both Temperature and Pressure at the same time, I've updated the state machine in devBaro to be able to skip the waiting for either a temperature or pressure result if the `getPressureDuration()` or `getTemperatureDuration()` returns 0 duration. This prevents the device from having to wait one loop period to do nothing and cuts the overall report time for this barometer from 100-102ms to 91-92ms.

### Hardware Caveats
With an external breakout board, RF output from the RX can cause a **tremendous** amount of noise in the measurement, making it virtually unusable. Here are some measurements taken with a SuperP RX at 100mW with the antennas within 1cm of the barometer. The values are millis(), temperature (C x100), pressure (hPa x10)
```
# Telemetry Off
130717 t=2573 p=101689
130809 t=2573 p=101689
130901 t=2573 p=101688
130993 t=2573 p=101688
131084 t=2573 p=101688
131176 t=2573 p=101688
131268 t=2573 p=101688
131360 t=2573 p=101688
131452 t=2573 p=101688
131544 t=2573 p=101688
131636 t=2573 p=101688
131728 t=2573 p=101688
131820 t=2573 p=101688
# Telemetry 1:2
575377 t=2569 p=101690
575469 t=2570 p=101692
575561 t=2317 p=66287
575653 t=2564 p=101692
575745 t=2568 p=101696
575837 t=2317 p=66287
575929 t=2566 p=101693
576021 t=2569 p=101693
[577114][E][Wire.cpp:513] requestFrom(): i2cRead returned Error 263
578115 t=-14140 p=115942
578153 t=2534 p=101691
578245 t=2545 p=101693
578337 t=2552 p=101692
578429 t=2317 p=66287
578521 t=2554 p=101691
578613 t=2559 p=101691
578705 t=2317 p=66287
578797 t=2564 p=101694
578889 t=2565 p=101693
578981 t=2566 p=101692
579073 t=2317 p=66287
```
As such, care should be taken to avoid RF interference if a user adds on an external BMP280. Even at normal telemetry rates, this **will** cause the data to be unstable and report VSpds of up to +/-20m/s.

### Tested 
* BetaFPV SuperP via CH12/CH13 I2C connector
* RadioMaster ER5A via CH4/CH5 output pins

IMG_20231120_123313
** Peak Jank Achieved! **

### CODE REVIEW

**Comment by CapnBry** [comment]
I added this to test with back when PK added the configurable I2C pins, but no current hardware uses it. If the rest of y'all aren't keen on adding external devices, I would not be crying myself to sleep on my huge pillow if this were not accepted.

Pros: any ELRS RX can get a barometer, and some RXes are even set up for external I2C devices
Cons: it's more code and you know how I hate more code!

**Comment by pkendall64** [comment]
I'm cool with it. I've ordered a BMP280 test with and it should be here shortly.

**Comment by pkendall64** [comment]
Tested on the SuperP, all good 👍 
IMG_0490

**Review by pkendall64 — APPROVED** [review]

File: src/lib/Baro/baro_bmp085.h
Comment (brandonrc) [inline src/lib/Baro/baro_bmp085.h:1]: Love the comments :)

**Comment by brandonrc** [comment]
Going to try and hunt down my BME280 to test this on today!

**Comment by brandonrc** [comment]
```
Hardware Caveats

With an external breakout board, RF output from the RX can cause a tremendous amount of noise in the measurement, making it virtually unusable. Here are some measurements taken with a SuperP RX at 100mW with the antennas within 1cm of the barometer. The values are millis(), temperature (C x100), pressure (hPa x10)
```
@CapnBry I am testing this on the 900mhz version of BetaFPV SuperP, where you using the 2.4 or 900? 

I have having issues with getting the rx to see my baro. Going to try it on a standalone dev board real quick to double check it is not my hardware first.

**Comment by brandonrc** [comment]
image

```
Hal Init
SX127x Reset
SX127x Ready!
SX127x Begin
RFAMP_hal Init
Detecting radio 1... 12 found! (match by REG_VERSION == 0x12
Detecting radio 2... 12 found! (match by REG_VERSION == 0x12
hwTimer Init
No baro detected
hwTimer interval: 10000
```

**Comment by brandonrc** [comment]
Hal Init
SX127x Reset
SX127x Ready!
SX127x Begin
RFAMP_hal Init
Detecting radio 1... 12 found! (match by REG_VERSION == 0x12
Detecting radio 2... 12 found! (match by REG_VERSION == 0x12
hwTimer Init
Detected baro: BMP280
1106 t=2555 p=99303
99303dPa 1583741cm
1198 t=2562 p=99303
99303dPa 1583741cm
diff=0 smooth=0 dT=91
1290 t=2570 p=99303
99303dPa 1583741cm

**Comment by CapnBry** [comment]
> @CapnBry Found the issue.. my address was 0x77 and not the standard 0x76

Ahhh fudge, I was worried it might have been this. I was looking last night and saw that the Adafruit sensor breakout uses 0x77 but all the breakouts I have use 0x76. Let me take a bit to rejigger the code to support both, which is a bit of an issue thanks to the address being a static member in the base class so the `detect()` method can't store which address the sensor was found on. Gonna have to throw that out I think...

**Comment by brandonrc** [comment]
> > @CapnBry Found the issue.. my address was 0x77 and not the standard 0x76
> 
> Ahhh fudge, I was worried it might have been this. I was looking last night and saw that the Adafruit sensor breakout uses 0x77 but all the breakouts I have use 0x76. Let me take a bit to rejigger the code to support both, which is a bit of an issue thanks to the address being a static member in the base class so the `detect()` method can't store which address the sensor was found on. Gonna have to throw that out I think...

Want me to rework this so it checks 76 and 77? Seems like that is common on all Baros even the non `280`s

**Comment by brandonrc** [comment]
I was fighting the static address... but I can look at the data sheet. I think all these baros have both options so maybe detect can be just a little more smart.. but then we would need to set the address if one fails right. try 77 set or try 76 then set?

**Comment by CapnBry** [comment]
Modified I2C base class to have static member that holds the detection address and detected address. Somehow works out to be 6 bytes less code, but 8 bytes more RAM. Tested with SPL06 and BME280.

**Review by schugabe — APPROVED** [review]

**Pull request merged**